src/edu/washington/escience/myria/perfenforce/PerfEnforceUtils.java

```diff
@@ -0,0 +1,179 @@
+   */
+  public static Set<Integer> getWorkerRangeSet(final int limit) {
+    Set<Integer> seq = new HashSet<Integer>();
+    for (int i = 1; i <= limit; i++) {
+      seq.add(i);
+    }
+    return seq;
+  }
+
+  /**
+   * Returns an array of worker ids
+   *
+   * @param min the min worker id (inclusive)
+   * @param max the max worker id (inclusive)
+   */
+  public static int[] getRangeInclusiveArray(final int min, final int max) {
+    int numberElements = (max - min) + 1;
+    int[] intArray = new int[numberElements];
+    for (int i = 0; i < numberElements; i++) {
+      intArray[i] = min + i;
+    }
+    return intArray;
+  }
+
+  /**
+   * Creates a UNION sql statement based on a set of relationKeys
+   *
+   * @param keysToUnion the relationKeys to union
+   */
+  public static String createUnionQuery(final List<RelationKey> keysToUnion) {
+    String sql = "";
+    for (RelationKey table : keysToUnion) {
+      sql +=
+          String.format(
+              "select * from \"%s:%s:%s\"",
+              table.getUserName(),
+              table.getProgramName(),
+              table.getRelationName());
+      if (table != keysToUnion.get(keysToUnion.size() - 1)) {
+        sql += " UNION ALL ";
+      }
+    }
+    return sql;
+  }
+
+  /**
+   * Concatenates an array of attributes in a string
+   *
+   * @param keys the set of columns to concatenate
+   * @param schema the schema of the relation
+   */
+  public static String getAttributeKeyString(final Set<Integer> keys, final Schema schema) {
+    String keyString = "";
+    int counter = 1;
+    for (int key : keys) {
+      keyString += schema.getColumnName(key);
+      if (counter != keys.size()) {
+        keyString += ",";
+      }
+      counter++;
+    }
+    return keyString;
+  }
+
+  /**
+   * Creates a new schema based on the given column ids
+   *
+   * @param keys the set of columns to concatenate
+   * @param schema the schema to create
+   */
+  public static Schema getAttributeKeySchema(final Set<Integer> keys, final Schema schema) {
+    Schema keySchema = Schema.EMPTY_SCHEMA;
+    for (int key : keys) {
+      keySchema =
+          Schema.appendColumn(keySchema, schema.getColumnType(key), schema.getColumnName(key));
+    }
+    return keySchema;
+  }
+
+  /**
+   * This method runs a SQL worker on a subset of workers. It will return the features with the highest cost to account for skew (if any)
+   *
+   * @param server an instance of the server
+   * @param sqlQuery the SQL statement to run
+   * @param configuration the configuration
+   * @throws PerfEnforceException
+   */
+  public static String getMaxFeature(final Server server, String sqlQuery, final int configuration)
+      throws PerfEnforceException {
+
+    try {
+      String explainQuery = "EXPLAIN " + sqlQuery;
+      List<String> featuresList = new ArrayList<String>();
+      String maxFeature = "";
+      double maxCost = Integer.MIN_VALUE;
```

Comment:
I'm pretty sure that the minimum cost for query execution from Postgres is 0.00. I can't find exact document to back this up, but I can't imagine there would be a query cost less than 0. In addition, I'm also only running SPJ queries. I'm not running indexes, stored procedures, or anything outside the ordinary :)